src/dodal/devices/attenuator_base.py

```diff
@@ -0,0 +1,15 @@
+from ophyd_async.core import StandardReadable
+from ophyd_async.epics.core import epics_signal_r, epics_signal_rw, epics_signal_x
+
+
+class AttenuatorBase(StandardReadable):
+    """An attenuator base class with a minimum set of common PVs to phase1 beamlines."""
+
+    def __init__(self, prefix: str, name: str = "") -> None:
+        self._desired_transmission = epics_signal_rw(float, prefix + "T2A:SETVAL1")
+        self._use_current_energy = epics_signal_x(prefix + "E2WL:USECURRENTENERGY.PROC")
```

Comment:
Should: Either this attenuator is only used for reading and we remove these PVs or we want to add the setting too, in which case `AttenuatorBase` should have a `set`. 

If we're going to do setting as well I actually think you might be able to get away with exactly the same `set` between the two. In the i24 case:
* `_filters_in_position` contains only `BL24I-OP-ATTN-01:MP1:INPOS` and `BL24I-OP-ATTN-01:MP2:INPOS`
* `_calculated_filter_states` are soft signals that are all high
*  `_change` is a soft signal that's a no-op

If we don't do this in this issue can you put this in a new issue please?

Reply:
Made an issue here #928